docs/batch-system-integration.md

```diff
@@ -0,0 +1,839 @@
+
+```hl_lines="5 10"
+JOB_ROUTER_ENTRIES @=jre
+[
+  TargetUniverse = 5;
+  name = "Limit the total number of idle jobs to 100";
+  MaxIdleJobs = 100;
+]
+[
+  TargetUniverse = 5;
+  name = "Limit the total number of idle jobs to 75";
+  MaxIdleJobs = 75;
+]
+@jre
+```
+
+### Debugging routes
+
+To help debug expressions in your routes, you can use the `debug()` function. First, set the debug mode for the JobRouter by editing a file in `/etc/condor-ce/config.d/` to read
+
+```
+JOB_ROUTER_DEBUG = D_ALWAYS:2 D_CAT
+```
+
+Then wrap the problematic attribute in `debug()`:
+
+```hl_lines="6"
+JOB_ROUTER_ENTRIES @=jre
+[
+  GridResource = "batch pbs";
+  TargetUniverse = 9;
+  name = "Debugging a difficult !ClassAd expression";
+  eval_set_Experiment = debug(strcat("atlas", Name));
+]
+@jre
+```
+
+You will find the debugging output in `/var/log/condor-ce/JobRouterLog`.
+
+Routes for HTCondor Batch Systems
+---------------------------------
+
+This section contains information about job routes that can be used if you are running an HTCondor batch system at your site.
+
+### Setting periodic hold, release or remove
+
+To release, remove or put a job on hold if it meets certain criteria, use the `PERIODIC_*` family of attributes. By default, periodic expressions are evaluated once every 300 seconds but this can be changed by setting `PERIODIC_EXPR_INTERVAL` in your CE's configuration.
+
+In this example, we set the routed job on hold if the job is idle and has been started at least once or if the job has tried to start more than once.  This will catch jobs which are starting and stopping multiple times.
+
+```hl_lines="6 10"
+JOB_ROUTER_ENTRIES @=jre
+[
+  TargetUniverse = 5;
+  name = "Setting periodic statements";
+  # Puts the routed job on hold if the job's been idle and has been started at least once or if the job has tried to start more than once
+  set_Periodic_Hold = (NumJobStarts >= 1 && JobStatus == 1) || NumJobStarts > 1;
+  # Remove routed jobs if their walltime is longer than 3 days and 5 minutes
+  set_Periodic_Remove = ( RemoteWallClockTime > (3*24*60*60 + 5*60) );
+  # Release routed jobs if the condor_starter couldn't start the executable and 'VMGAHP_ERR_INTERNAL' is in the HoldReason
+  set_Periodic_Release = HoldReasonCode == 6 && regexp("VMGAHP_ERR_INTERNAL", HoldReason);
+]
+@jre
+```
+
+### Setting routed job requirements
+
+If you need to set requirements on your routed job, you will need to use `set_Requirements` instead of `Requirements`. The `Requirements` attribute filters jobs coming into your CE into different job routes whereas `set_Requirements` will set conditions on the routed job that must be met by the worker node it lands on. For more information on requirements, consult the [HTCondor manual](http://research.cs.wisc.edu/htcondor/manual/v8.6/2_5Submitting_Job.html#SECTION00357000000000000000).
```

Comment:
```suggestion
If you need to set requirements on your routed job, you will need to use `set_Requirements` instead of `Requirements`. The `Requirements` attribute filters jobs coming into your CE into different job routes whereas `set_Requirements` will set conditions on the routed job that must be met by the worker node it lands on. For more information on requirements, consult the [HTCondor manual](https://htcondor.readthedocs.io/en/stable/users-manual/submitting-a-job.html#about-requirements-and-rank).
```